sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_func_utils.py

```diff
@@ -172,49 +246,102 @@ def _create_code(self, instructions: List[Instr], base_func: Union[FunctionType,
             fn_code.argnames.insert(0, self.injected_param)
             return fn_code.to_code()
 
+        @classmethod
+        def _get_mock_body_instructions(cls):
+            return [Instr("LOAD_FAST", "mock_arg")]
+
         # endregion
 
-        def _split_instructions(self, instructions, *, skip_body_instr=False) -> List[List[Any]]:
-            """Split instructions into several pieces by template separators.
-            For example, in Python 3.11, the template separators will be:
-            [
-                Instr('RESUME', 0),  # initial instruction shared by all functions
-                Instr('LOAD_FAST', 'mock_arg'),  # the body execution instruction
-                Instr('RETURN_VALUE'),  # the return instruction shared by all functions
-            ]
-            Then we will split the target template bytecode into 4 pieces.
-            For passed in bytecode, we should skip the body execution instruction, which is from template,
-            and split it into 3 pieces.
+        @classmethod
+        def _get_pieces(cls, _instructions, _separators):
+            """Split the instructions into pieces by the separators.
+            Return a list of pieces except the last piece and the last piece.
```

Comment:
except the **last piece** and the **last piece**?

Reply:
**a list of pieces except the last piece** and the **last piece**. Maybe I can add an "alone" at the end.

Reply:
updated the function to make its functionality easier to understand.